### DISCUSSION TITLE
Implementing FormCollection w/ Many-to-One Relation, Selectize Widget, and Dynamic Form Fields

### DISCUSSION THREAD

**Discussion by omarcoming** [jrief/django-formset#141] [Q&A] [answered]
Background:
I'm looking for a solution for the line item element for an invoice management system. The two models I'm using are Product and ProductDetail where each ProductDetail entry is associated with a Product.

Models:
- Product: Includes fields like unit, name, material, and vendor.
- ProductDetail: Includes fields like quantity, block number, width, length, price, and total, with each entry linked to a Product.

Requirement:
In the ProductDetail form, I want to implement the following functionality:
- Searchable Product Selection: The form should include a searchable field that allows users to select a product by its name using the Selectize widget. 
- Dynamic Field Population: Upon selecting a product, other associated Product fields (unit, material, vendor) should automatically populate and be displayed in a non-editable format.
- Additional Inputs: The form should also allow users to input values for the remaining ProductDetail fields (quantity, block number, width, length, price, total).

I went through all the documentation but I couldn't find a solution that matched my case. What would be the best way to implement this?

**Comment by jrief** [comment, marked as answer]
Hmm, interesting use case.

I see that django-formset is now being used in areas beyond its initial intention. I the future I might integrate it into frameworks such as django-unicorn or HTMX. There have been others asking for this already. Anyway, since each formset holds its own scope (always updated to latest current field values), this could be used as a context for client-side rendering templates.

Currently I don't have a short term solution for you. I would have to tinker myself around this.
